riscv/src/continuations.rs

```diff
@@ -141,23 +150,23 @@ fn sanity_check(program: &AnalysisASMFile) {
     }
 
     // Check that the registers of the machine are as expected.
-    let machine_registers = main_machine
+    let _machine_registers = main_machine
         .registers
         .iter()
         .filter_map(|r| {
             ((r.ty == RegisterTy::Pc || r.ty == RegisterTy::Write) && r.name != "x0")
                 .then_some(format!("main.{}", r.name))
         })
         .collect::<BTreeSet<_>>();
-    let expected_registers = REGISTER_NAMES
+    let _expected_registers = REGISTER_NAMES
         .iter()
         .map(|s| s.to_string())
         .collect::<BTreeSet<_>>();
     // FIXME: Currently, continuations don't support a Runtime with extra
     // registers. This has not been fixed because extra registers will not be
     // needed once we support accessing the memory machine from multiple
     // machines. This comment can be removed then.
-    assert_eq!(machine_registers, expected_registers);
+    //assert_eq!(machine_registers, expected_registers);
```

Comment:
I think we'll want this check?

Reply:
It's tricky. In order to change the bootloader code as little as possible, I kept the REGISTER_NAMES array as before, which includes all RISCV x* registers, tmp*, lc_sc_reservation, and pc. But machine registers will have only val1, val2, pc.

In the end, I think we actually don't need the check, because expected_registers is what the bootloader is concerned with, which are the regs mem addresses, which have nothing to do with machine registers. So maybe the whole block can be removed actually.

Reply:
Ah, right. I agree, could be removed!